src/DiagramGenerator/Diagram/Board.php

```diff
@@ -336,6 +337,29 @@ protected function getBackgroundTexture()
         return $boardCachedPath;
     }
 
+    protected function waitIfNotDownloaded($filePath)
+    {
+        $maxSleepTime = 100000; // tenth of a second
```

Comment:
I think that this approach is valid, I would check that:
- 1/10th of a second long enough?
- Waking up every 10µs doesn't cause too much load.  That's up to 10,000 wake-ups before it hits its hard limit.

Fewer wake-ups might be more efficient and just as effective.